test/compress/harmony.js

```diff
@@ -1212,3 +1212,36 @@ issue_2676: {
         (class {}).a = 42;
     }
 }
+
+issue_2762: {
+    mangle = {}
+    input: {
+        var bar = 1, T = true;
+        (function() {
+            if (T) {
+                const a = function() {
+                    var foo = bar;
+                    console.log(foo, a.prop, b.prop);
+                };
+                a.prop = 2;
+                const b = { prop: 3 };
+                a();
+            }
+        })();
+    }
+    expect: {
+        var bar = 1, T = true;
+        (function() {
+            if (T) {
+                const o = function() {
+                    var p = bar;
+                    console.log(p, o.prop, r.prop);
+                };
+                o.prop = 2;
+                const r = { prop: 3 };
+                o();
+            }
+        })();
+    }
+    expect_stdout: "1 2 3"
```

Comment:
```js
    node_version: ">=6"
```

Reply:
Heh - the code does work on Node.js 0.10 :ghost:

Reply:
Purely by accident I guess with the non-block scoped ES5 `const`.